ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureBlobDataAccessor.java

```diff
@@ -32,12 +36,15 @@
 import com.github.ambry.commons.BlobId;
 import com.github.ambry.config.CloudConfig;
 import com.github.ambry.utils.Utils;
+import com.google.common.collect.Lists;
```

Comment:
If we want to use guava directly in Ambry (I'm not against this since it has a lot of useful things), it might be good to explicitly reference the desired version in our build files. Interestingly, our dependencies seem to transitively bring in 3 versions of guava.

Reply:
If we decide not to use guava directly we can add a partition method to `Utils.java` (the impl is pretty simple), or just use `List::subList`

Reply:
Added Utils.partitionList() and refactored AzureCloudDestination.getBlobMetadata() to use it as well.